ambry-tools/src/main/java/com.github.ambry/clustermap/HelixBootstrapUpgradeTool.java

```diff
@@ -0,0 +1,466 @@
+   *                 {
+   *                   "datacenter":"dc1",
+   *                   "hostname":"abc.example.com",
+   *                   "port":"2199",
+   *                 },
+   *                 {
+   *                   "datacenter":"dc2",
+   *                   "hostname":"def.example.com",
+   *                   "port":"2300",
+   *                 }
+   *               ]
+   *             }
+   *
+   *             Also takes in an optional argument that specifies the local datacenter name, so that can be used as
+   *             the "reference" datacenter. If none provided, the tool simply chooses one of the datacenters in the
+   *             layout as the reference datacenter.
+   */
+  public static void main(String args[]) {
+    try {
+      OptionParser parser = new OptionParser();
+
+      ArgumentAcceptingOptionSpec<String> hardwareLayoutPathOpt =
+          parser.accepts("hardwareLayoutPath", "The path to the hardware layout json file")
+              .withRequiredArg()
+              .describedAs("hardware_layout_path")
+              .ofType(String.class);
+
+      ArgumentAcceptingOptionSpec<String> partitionLayoutPathOpt =
+          parser.accepts("partitionLayoutPath", "The path to the partition layout json file")
+              .withRequiredArg()
+              .describedAs("partition_layout_path")
+              .ofType(String.class);
+
+      ArgumentAcceptingOptionSpec<String> zkLayoutPathOpt = parser.accepts("zkLayoutPath",
+          "The path to the json file containing zookeeper connect info. This should"
+              + " be of the following form: \n{\n" + "  \"zkHosts\" : [\n" + "     {\n"
+              + "       \"datacenter\":\"dc1\",\n" + "       \"hostname\":\"abc.example.com\",\n"
+              + "       \"port\":\"2199\",\n" + "     },\n" + "     {\n" + "       \"datacenter\":\"dc2\",\n"
+              + "       \"hostname\":\"def.example.com\",\n" + "       \"port\":\"2300\",\n" + "     }\n" + "  ]\n"
+              + "}").
+          withRequiredArg().
+          describedAs("zk_connect_info_path").
+          ofType(String.class);
+
+      ArgumentAcceptingOptionSpec<String> localDcOpt = parser.accepts("localDc", "The local datacenter name")
+          .withRequiredArg()
+          .describedAs("local_dc")
+          .ofType(String.class);
+
+      OptionSet options = parser.parse(args);
+      String hardwareLayoutPath = options.valueOf(hardwareLayoutPathOpt);
+      String partitionLayoutPath = options.valueOf(partitionLayoutPathOpt);
+      String zkLayoutPath = options.valueOf(zkLayoutPathOpt);
+      ArrayList<OptionSpec> listOpt = new ArrayList<>();
+      listOpt.add(hardwareLayoutPathOpt);
+      listOpt.add(partitionLayoutPathOpt);
+      listOpt.add(zkLayoutPathOpt);
+      ToolUtils.ensureOrExit(listOpt, options, parser);
+      bootstrapOrUpgrade(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath, options.valueOf(localDcOpt));
+    } catch (Exception e) {
```

Comment:
this catch isn't required. Since this is the main function, it will simply exit the program and print the stack trace of the exception (the stack trace is lost if you print it this way).